### PR TITLE
ois: migrate to conan 2 and bump version to 1.5.1

### PR DESCRIPTION
**Pull request by joakimono**
Specify library name and version:  **ois/1.5**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ois/all/conandata.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 sources:
+  "1.5.1":
+    url: "https://github.com/wgois/OIS/archive/v1.5.1.tar.gz"
+    sha256: "614f6ef6d69cf6d84f1b50efff46a6c1acce426933e5f0dcf29862ea8332af73"
   "1.5":
     url: "https://github.com/wgois/OIS/archive/v1.5.tar.gz"
     sha256: "aa3e6b840b1149bb06835e04365957f4659c08b3e3095a48b289050f1637a174"
 patches:
+  "1.5.1": []
```
Comment (RubenRBS) [inline recipes/ois/all/conandata.yml:9]: If no patches are needed, no need for this key to be present
```suggestion
```

Is the patch for 1.5 now available in 1.5.1 then?

File: recipes/ois/all/conanfile.py [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir, collect_libs
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+
+required_conan_version = ">=1.53.0"
+
 
 class OisConan(ConanFile):
```
Comment (RubenRBS) [inline recipes/ois/all/conanfile.py:15]: Upstream defines CXX version of 11 https://github.com/wgois/OIS/blob/master/CMakeLists.txt#L6, so you'll want to add a `validate()` method to check it. [See this for an example on how to implement it](https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L78)

File: recipes/ois/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,7 @@
...
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+find_package(ois REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+target_link_libraries(${PROJECT_NAME} PRIVATE ois::ois)
```
Comment (RubenRBS) [inline recipes/ois/all/test_package/CMakeLists.txt:7]: Will also need to set the test package to C++ 11

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot for your contribution. The changes look mostly good to go, just a few minor things remaining :)

File: recipes/ois/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (joakimono) [reply recipes/ois/all/test_package/CMakeLists.txt:7]: Done

File: recipes/ois/all/conanfile.py (hunk 2 above)
Comment (joakimono) [reply recipes/ois/all/conanfile.py:15]: Done

**Review by RubenRBS — APPROVED** [review]

**Review by davidsanfal — APPROVED** [review]

**Pull request merged**